keps/sig-instrumentation/2305-metrics-cardinality-enforcement/README.md

```diff
@@ -236,19 +264,145 @@ This would then be interpreted by our machinery as this:
 
 ```
 
-## Open-Question
-
+## Design Details
+### Test Plan
+Unit test to verify that the metric label will be set to "unexpected" if the metric encounters label values outside our explicit allowlist of values.
+### Graduation Criteria
+The allowlist of metrics can be configured via the exposed flag and the unit test is passed.
+### Upgrade / Downgrade strategy
+N/A
```

Comment:
This almost certainly will need to go behind a feature flag, so this is applicable.

Reply:
About the feature flag here, do you mean it is another flag to enable metric cardinality rather than allow-list flag? I suppose it is no necessary because basically turning off allowlist flag will disable this feature.

Reply:
@johnbelamaric to clarify the requirements maybe, I'm not sure if a feature flag can be avoided for new CLI flags.

Reply:
I'm discussing with PRR team

Reply:
Yes, we need a feature gate. It allows users to know the state of the code.

Reply:
So the feature gate will basically affect all components that populate metrics, right?

Reply:
yeah, anywhere your flags would appear etc